schema_salad/tests/test_codegen_errors.py

```diff
@@ -0,0 +1,220 @@
+    path = get_data("tests/" + t)
+    assert path
+    with pytest.raises(ValidationException, match=match):
+        load_document_by_uri(path)
+
+
+def test_error_message6() -> None:
+    t = "test_schema/test6.cwl"
+    match = r"""-\s+tried\s+`CommandLineTool`\s+but
+\s+Missing\s+'class'\s+field
++-\s+tried\s+`ExpressionTool`\s+but
+\s+Missing\s+'class'\s+field
++-\s+tried\s+`Workflow`\s+but
+\s+Missing\s+'class'\s+field"""
+    path = get_data("tests/" + t)
+    assert path
+    with pytest.raises(ValidationException, match=match):
+        load_document_by_uri(path)
+
+
+def test_error_message7() -> None:
+    t = "test_schema/test7.cwl"
+    match = r"""^.*test7\.cwl:2:1:\s+Object\s+`.*test7\.cwl`\s+is\s+not\s+valid\s+because
+\s+tried\s+`Workflow`\s+but
+.*test7\.cwl:8:1:\s+the\s+`steps`\s+field\s+is\s+not\s+valid\s+because:
+\s+tried\s+`array<WorkflowStep>`\s+but
+.*test7\.cwl:9:3:\s+checking\s+object\s+`.*test7.cwl#step1`
+\s+tried\s+`WorkflowStep`\s+but
+\s+\*\s+missing\s+required\s+field\s+`run`
+.*test7\.cwl:10:5:\s+\*\s+invalid\s+field\s+`scatter_method`,\s+expected\s+one of:.*\s+.*\s+.*\s+.*\s+.*\s+.*\s+.*\s+.*\s+.*\s+.*$"""
+    path = get_data("tests/" + t)
+    assert path
+    with pytest.raises(ValidationException, match=match):
+        load_document_by_uri(path)
+
+
+def test_error_message8() -> None:
+    t = "test_schema/test8.cwl"
+    match = r"""^.*test8.cwl:2:1:\s+Object\s+`.*test8.cwl`\s+is\s+not\s+valid\s+because
+\s+tried\s+`Workflow`\s+but
+.*test8\.cwl:8:1:\s+the\s+`steps`\s+field\s+is\s+not\s+valid\s+because:
+\s+tried\s+`array<WorkflowStep>`\s+but
+.*test8\.cwl:9:3:\s+checking\s+object\s+`.*test8\.cwl#step1`
+\s+tried\s+`WorkflowStep`\s+but
+\s+\*\s+missing\s+required\s+field\s+`run`
+.*test8\.cwl:10:5:\s+the\s+`scatterMethod`\s+field\s+is\s+not\s+valid\s+because:
+\s+contains\s+undefined\s+reference\s+to
+\s+`file:///.*/tests/test_schema/abc`$"""
+    path = get_data("tests/" + t)
+    assert path
+    with pytest.raises(ValidationException, match=match):
+        load_document_by_uri(path)
+
+
+def test_error_message9() -> None:
+    t = "test_schema/test9.cwl"
+    match = r"""^.*test9.cwl:2:1:\s+Object\s+`.*test9.cwl`\s+is\s+not\s+valid\s+because
+\s+tried\s+`Workflow`\s+but
+.*test9\.cwl:8:1:\s+the\s+`steps`\s+field\s+is\s+not\s+valid\s+because:
+\s+tried\s+`array<WorkflowStep>`\s+but
+.*test9\.cwl:9:3:\s+checking object\s+`.*test9\.cwl#step1`
+\s+tried\s+`WorkflowStep`\s+but
+\s+\*\s+missing\s+required\s+field\s+`run`
+.*test9\.cwl:10:5:\s+the\s+`scatterMethod`\s+field\s+is\s+not\s+valid\s+because:
+\s+Expected\s+one\s+of\s+\("dotproduct|nested_crossproduct|flat_crossproduct",\s+"dotproduct|nested_crossproduct|flat_crossproduct",\s+"dotproduct|nested_crossproduct|flat_crossproduct"\) was\s+<class 'int'>"""
+    path = get_data("tests/" + t)
+    assert path
+    with pytest.raises(ValidationException, match=match):
+        load_document_by_uri(path)
+
+
+def test_error_message10() -> None:
```

Comment:
```
tests/test_schema/test10.cwl:2:1:   Object `test10.cwl` is not valid because
                                     tried `Workflow` but
tests/test_schema/test10.cwl:8:1:       the `steps` field is not valid because:
                                         tried `array<WorkflowStep>` but
tests/test_schema/test10.cwl:9:3:           checking object `test10.cwl#step1`
                                             tried `WorkflowStep` but
                                               * missing required field `run`
tests/test_schema/test10.cwl:10:5:             the `scatterMethod` field is not valid because:
                                                 Expected one of ("flat_crossproduct",
                                                 "nested_crossproduct", "dotproduct") was <class
                                                 'ruamel.yaml.comments.CommentedSeq'>
```

Original validator:

```
test_schema/test10.cwl:2:1:   Object 'test_schema/test10.cwl' is not valid because
                               tried 'Workflow' but
test_schema/test10.cwl:8:1:       the 'steps' field is not valid because
                                   tried array of <WorkflowStep> but
test_schema/test10.cwl:9:3:           item is invalid because
                                       * missing required field 'run'
test_schema/test10.cwl:10:5:           * the 'scatterMethod' field is not valid because
                                         value is a CommentedSeq, expected null or ScatterMethod
```

* Same issue I mentioned on test8 with the bullet points
* I know the original validator is doing the same thing, but we should recognize when an object is "CommentedSeq" and just call it "list"